packages/core/src/types/ad.ts

```diff
@@ -39,9 +39,22 @@ import { LbfgsParams } from "./state";
 
 //#region Types for implicit autodiff graph
 
-export type VarAD = Const | Input | Unary | Binary | Ternary | Nary | Debug;
+export type Expr = Bool | VarAD | Vec;
 
-export type Const = ConstNode;
+export type Bool = Comp | Logic;
+
+// this type name is a relic, doesn't follow our current convention
```

Comment:
Perhaps rename to `ad.Var`?

Reply:
I'm fine with doing that in a followup PR that just does the rename, but I want to keep this diff small and focused.

Reply:
Although if we do rename, I'd want to call it `ad.Num` instead of `ad.Var`.

Reply:
Yeah that sounds great!